src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/AddColumns.cs

```diff
@@ -78,41 +77,68 @@ public override bool CheckTypes(TexlNode[] args, DType[] argTypes, IErrorContain
                 errors.EnsureError(DocumentErrorSeverity.Severe, args[0].Parent.CastList().Parent.CastCall(), TexlStrings.ErrBadArityOdd, count);
             }
 
+            var supportColumnNamesAsIdentifiers = context.Features.HasFlag(Features.SupportColumnNamesAsIdentifiers);
+
             for (var i = 1; i < count; i += 2)
             {
+                string expectedColumnName = null;
                 var nameArg = args[i];
                 var nameArgType = argTypes[i];
 
                 // Verify we have a string literal for the column name. Accd to spec, we don't support
                 // arbitrary expressions that evaluate to string values, because these values contribute to
                 // type analysis, so they need to be known upfront (before AddColumns executes).
-                StrLitNode nameNode;
-                if (nameArgType.Kind != DKind.String ||
-                    (nameNode = nameArg.AsStrLit()) == null)
+                if (supportColumnNamesAsIdentifiers)
                 {
-                    fArgsValid = false;
-                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
-                    continue;
+                    if (nameArg is not FirstNameNode identifierNode)
+                    {
+                        fArgsValid = false;
+
+                        // Argument '{0}' is invalid, expected an identifier.
+                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedIdentifierArg_Name, nameArg.ToString());
+                        continue;
+                    }
+
+                    expectedColumnName = identifierNode.Ident.Name;
+                }
+                else
+                {
+                    StrLitNode strLitNode = nameArg.AsStrLit();
+
+                    if (nameArgType.Kind != DKind.String && strLitNode == null)
```

Comment:
>&& [](http://example.com/codeflow?start=57&length=2)

This should be an Or `||` condition.

Reply:
Otherwise we would miss error and throw null pointer on arg which is FirstName node with string type.
since in that case, `nameArgtype.Kind != DKind.String` would be `false`

Reply:
Good eyes!